src/engine/editor/map_config_editor.cpp

```diff
@@ -37,6 +37,101 @@ MapConfigEditor::~MapConfigEditor() {
     RestoreManifest();
 }
 
+void MapConfigEditor::DisplayTemperature() 
+{
+    enum {
+        TEMP_NORMAL, TEMP_HOT, TEMP_COLD, MAX_TEMP
+    };
+    static const char *temperatures[MAX_TEMP] = {"Normal", "Hot", "Cold"};
+    static int temperature_index = -1;
+    if (temperature_index == -1) {
+        if (pl_strcasecmp(manifest_->temperature.c_str(), "Hot") == 0) {
+            temperature_index = TEMP_HOT;
+        } else  if (pl_strcasecmp(manifest_->temperature.c_str(), "Cold") == 0) {
+            temperature_index = TEMP_COLD;
+        } else {
+            temperature_index = TEMP_NORMAL;
+        }
+    }
+
+    if (ImGui::BeginCombo("Temperature", temperatures[temperature_index])) {
+        for (int i = 0; i < MAX_TEMP; ++i) {
+            if (ImGui::Selectable(temperatures[i], (temperature_index == i))) {
+                ImGui::SetItemDefaultFocus();
+                temperature_index = i;
+
+                // TODO: convert to lower case
+                manifest_->temperature = std::string(temperatures[temperature_index]);
+            }
+        }
+
+        ImGui::EndCombo();
+    }
+}
+
+void MapConfigEditor::DisplayWeather()
+{
+    enum {
+        WEATHER_CLEAR, WEATHER_RAIN, WEATHER_SNOW, MAX_WEATHER
+    };
+    static const char *weatherTypes[MAX_WEATHER] = {"Clear", "Rain", "Snow"};
+    static int weather_index = -1;
+    if (weather_index == -1) {
+        if (pl_strcasecmp(manifest_->weather.c_str(), "Rain") == 0) {
+            weather_index = WEATHER_RAIN;
+        }
+        else if (pl_strcasecmp(manifest_->weather.c_str(), "Snow") == 0) {
+            weather_index = WEATHER_SNOW;
+        } else {
+            weather_index = WEATHER_CLEAR;
+        }
+    }
+
+    if (ImGui::BeginCombo("Weather", weatherTypes[weather_index])) {
+        for (int i = 0; i < MAX_WEATHER; ++i) {
+            if (ImGui::Selectable(weatherTypes[i], (weather_index == i))) {
+                ImGui::SetItemDefaultFocus();
+                weather_index = i;
+
+                // TODO: convert to lower case
+                manifest_->weather = std::string(weatherTypes[weather_index]);
+            }
+        }
+
+        ImGui::EndCombo();
+    }
+}
+
+void MapConfigEditor::DisplayTime()
+{
+    enum {
+        TIME_DAY, TIME_NIGHT, MAX_TIME
+    };
+    static const char *times[MAX_TIME] = {"Day", "Night"};
+    static int time_index = -1;
+    if (time_index == -1) {
+        if (pl_strcasecmp(manifest_->time.c_str(), "Day") == 0) {
+            time_index = TIME_DAY;
+        } else {
+            time_index = TIME_NIGHT;
+        }
+    }
+
+    if (ImGui::BeginCombo("Time", times[time_index])) {
+        for (int i = 0; i < MAX_TIME; ++i) {
+            if (ImGui::Selectable(times[i], (time_index == i))) {
+                ImGui::SetItemDefaultFocus();
+                time_index = i;
+
+                // TODO: convert to lower case
+                manifest_->time = std::string(times[time_index]);
```

Comment:
The std::string is unnecessary here. 

You could utilise pl_strtolower, but you will need to create a char buffer and copy the times[time_index] result into it before you can pass it into the pl_strtolower function. It would be a little ugly...

Otherwise we should look at implementing a tolower function for std string specifically (this isn't too difficult at all).

A simpler and arguably lazy solution might be to just have a second array.

static const char *times_labels[MAX_TIME] = {"Day", "Night"};
static const char *times[MAX_TIME] = {"day", "night"};

I don't really have a preference, but I would prefer the values added to the manifest are lowercase regardless.